powersimdata/input/converter/pypsa_to_profiles.py

```diff
@@ -1,33 +1,69 @@
+def get_pypsa_gen_profile(network, profile2carrier):
     """Return hydro, solar or wind profile enclosed in a PyPSA network.
 
     :param pypsa.Network network: the Network object.
-    :param str kind: either *'hydro'*, *'solar'*, *'wind'*.
-    :return: (*pandas.DataFrame*) -- profile.
+    :param dict profile2carrier: a dictionary mapping profile type to carrier type.
+        *'hydro'*, *'solar'* and *'wind'* are valid keys. Values is a corresponding
+        set of carriers as found in the Network object.
+    :return: (*dict*) -- keys are the same ones than in ``profile2carrier``. Values
+        are profiles as data frame.
+    :raises TypeError:
+        if ``network`` is not a pypsa.components.Network object.
+        if ``profile2carrier`` is not a dict.
+        if values of ``profile2carrier`` are not an iterable.
+    :raises ValueError:
+        if keys of ``profile2carrier`` are invalid.
     """
-    p_max_pu = get_switchable_as_dense(network, "Generator", "p_max_pu")
-    p_max_pu.columns = pd.to_numeric(p_max_pu.columns, errors="ignore")
-    p_max_pu.columns.name = None
-    p_max_pu.index.name = "UTC"
+    if not isinstance(network, pypsa.components.Network):
+        raise TypeError("network must be a Network object")
+    if not isinstance(profile2carrier, dict):
+        raise TypeError("profile2carrier must be a dict")
+    if not all(isinstance(v, (list, set, tuple)) for v in profile2carrier.values()):
+        raise TypeError("values of profile2carrier must be an iterable")
+    if not set(profile2carrier).issubset({"hydro", "solar", "wind"}):
+        raise ValueError(
+            "keys of profile2carrier must be a subset of ['hydro', 'solar', 'wind']"
+        )
 
-    all_gen = network.generators.copy()
-    all_gen.index = pd.to_numeric(all_gen.index, errors="ignore")
-    all_gen.index.name = None
+    profile = {}
+    for p, c in profile2carrier.items():
+        profile[p] = pd.DataFrame()
+        for component in ["generators", "storage_units", "stores"]:
+            if hasattr(network, component):
+                carrier_in_component = getattr(network, component)
+                id_carrier = set(carrier_in_component.query("carrier==list(@c)").index)
+                for t in ["inflow", "p_max_pu"]:
+                    if t in getattr(network, component + "_t"):
+                        ts = getattr(network, component + "_t")[t]
+                        if not ts.empty:
+                            id_ts = set(ts.columns)
+                            idx = list(id_carrier.intersection(id_ts))
+                            norm = ts[idx].max() if t == "inflow" else 1
+                            profile[p] = pd.concat([profile[p], ts[idx] / norm], axis=1)
+                if len(set(c) - set(carrier_in_component.carrier.unique())):
+                    continue
+                else:
+                    break
+        profile[p].rename_axis(index="UTC", columns=None, inplace=True)
```

Comment:
I am not sure whether this code is a bit bloated. 
Pypsa's `get_switchable_as_dense` can also be used with 'Store' and 'StorageUnit' and the attribute 'inflow'. Does that make this easier here?

Reply:
I guess I could use `get_switchable_as_dense` to access the profile. I believe I will have to use a `try`/`except` logic since depending on the `Network` object, hydro/PHS can be located in the the `SorageUnit` component or `Store` component. Is that correct? If we focus on PyPSA-Eur only, the code an be greatly simplified. I was trying to be general.

Reply:
Right, a hydro/PHS facility could theoretically be a Store component. But it is quite unlikely, since the Store component does not have an "inflow" profile, only the StorageUnit component has. So I'd say for the inflow profile at least, it is not necessary to access stores.

Reply:
Thanks for the explanation. I will simplify the logic then.